schema/check_test_output.py

```diff
@@ -77,9 +77,21 @@ def main(args):
     schema_data_results = []
     schema_count = 0
 
-    all_results = validator.validate_test_output_with_schema()
+    all_results, test_validation_plans = validator.validate_test_output_with_schema()
     logging.info('  %d results for test output', len(all_results))
 
+    # Check if any files in the expected list were not validated.
+    test_paths = []
+    for plan in test_validation_plans:
+        test_paths.append(plan['test_result_file'])
+    unvalidated_json = []
+    for json_file in json_files:
+        if json_file not in test_paths:
+            logging.warning('JSON file %s was not verified against a schema', json_file)
+            unvalidated_json.append(json_file)
+    if unvalidated_json:
+        logging.warning(' %s json files not validated', len(unvalidated_json))
```

Comment:
All we're doing currently is warning. But having something that is not validated against a schema seems serious enough that it should be an error level log event, or maybe even a fatal level log event (fatal = error + quit immediately).

If we intend to ensure that everything we consume and produce is validated against schemas, then we should make this `logging.fatal` and quit immediately.

Reply:
Yup!